x-pack/elastic-agent/pkg/core/monitoring/beats/beats_monitor.go

```diff
@@ -122,8 +122,12 @@ func (b *Monitor) EnrichArgs(spec program.Spec, pipelineID string, args []string
 		appendix = append(appendix,
 			"-E", "http.enabled=true",
 			"-E", "http.host="+endpoint,
-			"-E", "http.pprof.enabled=true",
 		)
+		if b.config.Pprof {
```

Comment:
I know this is the way it works today but it is really ugly that we pass these things via cmd line params ...

Reply:
Yeah, we should change this in the future